.github/workflows/e2e.yml

```diff
@@ -258,16 +202,8 @@ jobs:
     timeout-minutes: 10
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.8.17
```

Comment:
Why is the default for the new bootstrap action 3.8.16 if .17 is available?

Reply:
I'll update it to .17. 

As a sidenote in the improve_ci_speed PR there will be some more changes and explanations on that.